datasets/pubmed/pubmed.py

```diff
@@ -0,0 +1,399 @@
+# coding=utf-8
+# Copyright 2020 The HuggingFace Datasets Authors and the current dataset script contributor.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""TODO: Add a description here."""
+
+from __future__ import absolute_import, division, print_function
+
+import copy
+import logging
+import xml.etree.ElementTree as etree
+
+import datasets
+
+
+logger = logging.getLogger(__name__)
+
+
+# Find for instance the citation on arxiv or on the dataset repo/website
+_CITATION = """\
+"""
+
+# You can copy an official description
+_DESCRIPTION = """\
+NLM produces a baseline set of MEDLINE/PubMed citation records in XML format for download on an annual basis. The annual baseline is released in December of each year. Each day, NLM produces update files that include new, revised and deleted citations. See our documentation page for more information.
+"""
+
+_HOMEPAGE = "https://www.nlm.nih.gov/databases/download/pubmed_medline.html"
+
+_LICENSE = ""
+
+# TODO: Add link to the official dataset URLs here
+# The HuggingFace dataset library don't host the datasets but only point to the original files
+# This can be an arbitrary nested dict/list of URLs (see below in `_split_generators` method)
+_URLs = [f"ftp://ftp.ncbi.nlm.nih.gov/pubmed/baseline/pubmed21n{i:04d}.xml.gz" for i in range(1, 1063)]
```

Comment:
Because there are 1000+ urls here, then currently 1000+ files are expected to be in the dummy_data.zip file as well. This is a bit overkill imo and it makes the zip file unnecessarily big (1MB+).

There is a way to only have 1 file in the dummy zip file used for tests when urls are formatted in a certain way.
Currently it only works for shards formatted using the tfrecords convention (for example `data.txt.0001-of-0300`) but feel free to extend it so that it also supports the urls formats that you have here.

You can see the current regex here: https://github.com/huggingface/datasets/blob/master/src/datasets/utils/mock_download_manager.py#L173-L175

Basically if the input of the dl_manager is a list of url that matches a regex, then during the dummy data tests the output is going to be a list of path to one file (instead of 1000+ different files).

Reply:
Ok.
I have done it with a similar approach though it might seem fair to have a different hook if this becomes regular :)